Fix grammar and typos (#953)
solidjs/solid-docs: src/routes/reference/basic-reactivity/create-resource.mdx

#### v1.5.0

1. We've added a new state field which covers a more detailed view of the Resource state beyond `loading` and `error`. 

> You can now check whether a Resource is `unresolved`, `pending`, `ready`, `refreshing`, or `errored`.

Reviewer: ```suggestion
You can now check whether a Resource is `unresolved`, `pending`, `ready`, `refreshing`, or `error`.
```

| State        | Value resolved | Loading | Has error |
| ------------ | -------------- | ------- | --------- |
| `unresolved` | No             | No      | No        |
| `pending`    | No             | Yes     | No        |
| `ready`      | Yes            | No      | No        |
| `refreshing` | Yes            | Yes     | No        |

> | `errored`      | No             | No      | Yes       |

Reviewer: ```suggestion
| `error`      | No             | No      | Yes       |
```

| `refreshing` | Yes            | Yes     | No        |
| `error`      | No             | No      | Yes       |

> 2. When server rendering resources especially when fetching when embedding Solid in other systems that fetch data before rendering, you might want to initiate the resource with this prefetched value instead of fetching again and having the resource serialize it in it's own state.

Reviewer 2: ```suggestion
2. When server-rendering resources, especially when embedding Solid in other systems that fetch data before rendering, you might want to initialize the resource with this prefetched value instead of fetching again and having the resource serialize it in its own state.
```

(Your changes here are good, but reading the sentence, I spotted some other fixes that would help.)
